askama_parser/src/lib.rs

```diff
@@ -487,3 +505,43 @@ fn filter(i: &str, level: Level) -> ParseResult<'_, (&str, Option<Vec<Expr<'_>>>
     ))(i)?;
     Ok((i, (fname, args)))
 }
+
+/// The goal of this function is to reduce the path length based on the current directory
+/// the programming is running into. For example:
+///
+/// ```text
+/// current dir: /a/b/c
+/// path:        /a/b/c/d/e.txt
+/// ```
+///
+/// `strip_common` will return `d/e.txt`.
+fn strip_common(path: &Path) -> String {
```

Comment:
I'd like to see a bunch of unit tests for this in a `tests` module at the bottom of the file. As such, we probably want to have the caller call in a `cwd`.

Reply:
I can generate paths based on the cwd to make things simpler. I'll add tests to enforce it. :+1:

Reply:
My suggestion was to extract the call to `std::env::current_dir()` from `strip_common()`, having the caller pass in a `&Path` containing the `cwd`.

Reply:
Oh I see. That will make tests easier indeed.